core/src/test/scala/unit/kafka/server/BrokerRegistrationRequestTest.scala

```diff
@@ -42,12 +43,15 @@ import java.util.concurrent.{CompletableFuture, TimeUnit, TimeoutException}
 class BrokerRegistrationRequestTest {
 
   def brokerToControllerChannelManager(clusterInstance: ClusterInstance): NodeToControllerChannelManager = {
+    def anyControllerSocketServer(clusterInstance: ClusterInstance): SocketServer = {
+      clusterInstance.controllers().values().stream().map(_.socketServer).findFirst().get()
+    }
     new NodeToControllerChannelManagerImpl(
       new ControllerNodeProvider() {
         def node: Option[Node] = Some(new Node(
-          clusterInstance.anyControllerSocketServer().config.nodeId,
+          anyControllerSocketServer(clusterInstance).config.nodeId,
           "127.0.0.1",
-          clusterInstance.anyControllerSocketServer().boundPort(clusterInstance.controllerListenerName()),
+          anyControllerSocketServer(clusterInstance).boundPort(clusterInstance.controllerListenerName()),
```

Comment:
this looks like a bug. It could create a nonexistent node, right? the port and broker id could be unmatched